backbone.js

```diff
@@ -750,8 +753,12 @@
           options.index = index;
           model.trigger('remove', model, this, options);
         }
+        models[j++] = model;
         this._removeReference(model, options);
       }
+      // We only need to slice if models array should be smaller, which is
+      // caused by some models not actually getting removed.
+      if (models.length !== j) models = models.slice(0, j);
       return singular ? models[0] : models;
```

Comment:
Hmm I'm not sure. The loop above is looping over `models`, which would only have 1 thing so I don't see how it could remove more than 1 model. Can you clarify? I might just be missing something. Plus, you can't have the same model in the collection more than once, right?